gossip/infostore.go

```diff
@@ -200,7 +209,7 @@ func (is *infoStore) registerCallback(pattern string, method Callback) {
 	// Run callbacks in a goroutine to avoid mutex reentry.
 	go func() {
 		for key, i := range infos {
-			method(key, true /* contentsChanged */, i.Val)
+			method(key, true /* contentsChanged */, i.Value.Bytes)
```

Comment:
After reading through the full pr, I can see what you mean.  But please see my comment below.